src/runtime_src/core/tools/xbmgmt2/xbmgmt

```diff
@@ -1,29 +1,84 @@
 #!/bin/bash
 
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2020-2021 Xilinx, Inc.
+#
+
 # -- Detect a Windows environment and automatically switch to the .bat file
 if [[ "`uname`" == windows32* ]] || [[ "`uname`" == CYGWIN* ]] ; then
   trap "" INT
   "$0.bat" "$@"
   exit $?
 fi
 
+DEFAULT=xbmgmt
+XBMGMT=xbmgmt
+XBMGMT2=xbmgmt2
+
+# Table to resolve reset sub-command based on switch passed to reset
+declare -A RSTOPT=( \
```

Comment:
I like what you did with the COMMANDS map below where you used the XBMGMT, XBMGM2, XBUTIL, and XBUTIL2 variables instead of the actual values like in RSTOPT, for focuses on the variable's name and not value (like this mapping does).  

While it is highly unlikely that the underlying command name will change in the future, by using the variable's name and not value does make for a more scalable solution.

Reply:
Updated